TACC/Core-Portal
# task/FP-1639: Delegate management

## Overview
Adds ability for TAS project PIs to designate and un-designate a user as a delegate.


## Related

* [FP-1639](https://jira.tacc.utexas.edu/browse/FP-1639)

## Changes

- Also adds route-aware state for the view team modal

## Testing

1. As a project PI, go to https://cep.dev/workbench/allocations/approved
2. Click "View Team" on a project you are a PI on, and head to the "Manage Team" tab.
3. Use the dropdown selector to alter the role of one of the project members, and click "Update".
4. Confirm the resulting roles are what you intended, even on page refresh.
5. (Also, refresh the page with the "View Team" modal open, and confirm the modal opens back up at that team.)
6. Confirm you cannot change the PIs role, nor any other users roles on a project you are not a PI of.

## UI

https://user-images.githubusercontent.com/20326896/182717013-103a0ae8-0be0-44fe-bddb-5faf1accef86.mov

![Screen Shot 2022-08-03 at 4 32 31 PM](https://user-images.githubusercontent.com/20326896/182717033-72ecd2f7-8e25-43ae-8b55-ad1615d2b8fe.png)
